pennylane/qnode_new/decorator.py

```diff
@@ -87,12 +108,20 @@ def QNode(
     return node

```

Comment:
I feel like this QNode creation got somewhat confusing lately. I did not follow the exact changes you made, but as far as I can tell from the code each new `QNode` subclass is basically embodying a different differentiation rule. But why aren't they then named `QubitParameterShiftQNode`, `CVParameterShiftQNode` or `FiniteDifferencesQNode`? Do you have an ADR you can send me detailing those QNode refactors?

Reply:
I agree, I originally named them `QubitParameterShiftQNode` but panicked at the last minute and changed them back :laughing: 

The idea being that we can implement new gradient rules via new QNodes. But since we only have parameter-shift for now, I shortened it to `QubitQNode` for the time being.

Reply:
(@co9olguy)